extensions/windows-event-log/wel/WindowsEventLog.h

```diff
@@ -83,76 +80,50 @@ class WindowsEventLogHandler {
 class WindowsEventLogMetadata {
  public:
   virtual ~WindowsEventLogMetadata() = default;
-  [[nodiscard]] virtual std::string getEventData(EVT_FORMAT_MESSAGE_FLAGS flags) const = 0;
+  [[nodiscard]] virtual nonstd::expected<std::string, std::error_code> getEventData(EVT_FORMAT_MESSAGE_FLAGS flags) const noexcept = 0;
   [[nodiscard]] virtual std::string getEventTimestamp() const = 0;
   virtual short getEventTypeIndex() const = 0;  // NOLINT short comes from WINDOWS API
 
-  static std::string getMetadataString(METADATA val) {
-    static std::map<METADATA, std::string> map = {
-        {LOG_NAME, "LOG_NAME"},
-        {SOURCE, "SOURCE"},
-        {TIME_CREATED, "TIME_CREATED"},
-        {EVENTID, "EVENTID"},
-        {OPCODE, "OPCODE"},
-        {EVENT_RECORDID, "EVENT_RECORDID"},
-        {EVENT_TYPE, "EVENT_TYPE"},
-        {TASK_CATEGORY, "TASK_CATEGORY"},
-        {LEVEL, "LEVEL"},
-        {KEYWORDS, "KEYWORDS"},
-        {USER, "USER"},
-        {COMPUTER, "COMPUTER"}
-    };
-
-    return map[val];
-  }
-
-  static METADATA getMetadataFromString(const std::string& val) {
-    static std::map<std::string, METADATA> map = {
-        {"LOG_NAME", LOG_NAME},
-        {"SOURCE", SOURCE},
-        {"TIME_CREATED", TIME_CREATED},
-        {"EVENTID", EVENTID},
-        {"OPCODE", OPCODE},
-        {"EVENT_RECORDID", EVENT_RECORDID},
-        {"TASK_CATEGORY", TASK_CATEGORY},
-        {"EVENT_TYPE", EVENT_TYPE},
-        {"LEVEL", LEVEL},
-        {"KEYWORDS", KEYWORDS},
-        {"USER", USER},
-        {"COMPUTER", COMPUTER}
-    };
-
-    auto enumVal = map.find(val);
-    if (enumVal != std::end(map)) {
-      return enumVal->second;
-    } else {
-      return METADATA::UNKNOWN;
-    }
+  static Metadata getMetadataFromString(std::string_view val) {
+    using enum Metadata;
+    if (val == "LOG_NAME") { return LOG_NAME; }
+    else if (val == "SOURCE") { return SOURCE; }
+    else if (val == "TIME_CREATED") { return TIME_CREATED; }
+    else if (val == "EVENTID") { return EVENTID; }
+    else if (val == "OPCODE") { return OPCODE; }
+    else if (val == "EVENT_RECORDID") { return EVENT_RECORDID; }
+    else if (val == "EVENT_TYPE") { return EVENT_TYPE; }
+    else if (val == "TASK_CATEGORY") { return TASK_CATEGORY; }
+    else if (val == "LEVEL") { return LEVEL; }
+    else if (val == "KEYWORDS") { return KEYWORDS; }
+    else if (val == "USER") { return USER; }
+    else if (val == "COMPUTER") { return COMPUTER; }
+    else { return UNKNOWN; }
   }
 
   static std::string getComputerName() {
-    static std::string computer_name;
-    if (computer_name.empty()) {
-      char buff[10248];
+    static std::string computer_name = []() -> std::string {
```

Comment:
Moved this to an immediately invoked lambda: this way the thread-safe static initialization takes care of only running the initialization once, and subsequent calls will just copy the static string. The old version was not thread-safe.